crates/re_viewer/src/misc/transform_cache.rs

```diff
@@ -236,70 +252,30 @@ fn transform_at(
                 } else {
                     *encountered_pinhole = true;
 
-                    // A pinhole camera means that we're looking at an image.
-                    // Images are spanned in their local x/y space.
-                    // Center it and move it along z, scaling the further we move.
-                    let props = entity_properties.get(entity_path);
-                    let distance = *props.pinhole_image_plane_distance.get();
+                    // A pinhole camera means that we're looking at some 2D image plane from a single point (the pinhole).
+                    // Center the image plane and move it along z, scaling the further the image plane is.
+                    let distance = pinhole_image_plane_distance(entity_path);
 
-                    let focal_length = pinhole.focal_length_in_pixels();
-                    let focal_length = glam::vec2(focal_length.x(), focal_length.y());
-                    let scale = distance / focal_length;
+                    let scale = distance / pinhole.focal_length_in_pixels();
                     let translation = (-pinhole.principal_point() * scale).extend(distance);
-                    let parent_from_child = glam::Mat4::from_scale_rotation_translation(
-                        // We want to preserve any depth that might be on the pinhole image.
-                        // Use harmonic mean of x/y scale for those.
-                        scale.extend(1.0 / (1.0 / scale.x + 1.0 / scale.y)),
+                    let parent_from_child = glam::Affine3A::from_scale_rotation_translation(
+                        glam::Vec3::splat(scale),
                         glam::Quat::IDENTITY,
                         translation,
                     );
```

Comment:
probably faster to use `Affine3A::from_translation(translation) * Affine3A::from_scale(scale)` to not have to convert a Quat to a Mat3